shell/platform/android/io/flutter/embedding/android/FlutterFragmentActivity.java

```diff
@@ -0,0 +1,671 @@
+     * following property: {@code <item name="android:windowIsTranslucent">true</item>}.
+     */
+    @NonNull
+    public CachedEngineIntentBuilder backgroundMode(@NonNull BackgroundMode backgroundMode) {
+      this.backgroundMode = backgroundMode.name();
+      return this;
+    }
+
+    /**
+     * Creates and returns an {@link Intent} that will launch a {@code FlutterFragmentActivity} with
+     * the desired configuration.
+     */
+    @NonNull
+    public Intent build(@NonNull Context context) {
+      return new Intent(context, activityClass)
+          .putExtra(EXTRA_CACHED_ENGINE_ID, cachedEngineId)
+          .putExtra(EXTRA_DESTROY_ENGINE_WITH_ACTIVITY, destroyEngineWithActivity)
+          .putExtra(EXTRA_BACKGROUND_MODE, backgroundMode);
+    }
+  }
+
+  @Nullable
+  private FlutterFragment flutterFragment;
+
+  @Override
+  protected void onCreate(@Nullable Bundle savedInstanceState) {
+    switchLaunchThemeForNormalTheme();
+
+    super.onCreate(savedInstanceState);
+
+    configureWindowForTransparency();
+    setContentView(createFragmentContainer());
+    configureStatusBarForFullscreenFlutterExperience();
+    ensureFlutterFragmentCreated();
+  }
+
+  /**
+   * Switches themes for this {@code Activity} from the theme used to launch this
+   * {@code Activity} to a "normal theme" that is intended for regular {@code Activity}
+   * operation.
+   * <p>
+   * This behavior is offered so that a "launch screen" can be displayed while the
+   * application initially loads. To utilize this behavior in an app, do the following:
+   * <ol>
+   *   <li>Create 2 different themes in style.xml: one theme for the launch screen and
+   *   one theme for normal display.
+   *   <li>In the launch screen theme, set the "windowBackground" property to a {@code Drawable}
+   *   of your choice.
+   *   <li>In the normal theme, customize however you'd like.
+   *   <li>In the AndroidManifest.xml, set the theme of your {@code FlutterFragmentActivity} to
+   *   your launch theme.
+   *   <li>Add a {@code <meta-data>} property to your {@code FlutterFragmentActivity} with a name
+   *   of "io.flutter.embedding.android.NormalTheme" and set the resource to your normal
+   *   theme, e.g., {@code android:resource="@style/MyNormalTheme}.
+   * </ol>
+   * With the above settings, your launch theme will be used when loading the app, and
+   * then the theme will be switched to your normal theme once the app has initialized.
+   * <p>
+   * Do not change aspects of system chrome between a launch theme and normal theme. Either define
+   * both themes to be fullscreen or not, and define both themes to display the same status bar and
+   * navigation bar settings. If you wish to adjust system chrome once your Flutter app renders, use
+   * platform channels to instruct Android to do so at the appropriate time. This will avoid any
+   * jarring visual changes during app startup.
+   */
+  private void switchLaunchThemeForNormalTheme() {
```

Comment:
Are some of these below copy pasted from the FlutterActivity file? Could some of it be extracted into a shared class?